validation-script/validator.py

```diff
@@ -0,0 +1,127 @@
+# recordType
+# (not included) request
+# sanitized
+# schemaVersion
+# securityResultCode
+# serialId
+#   streamId
+#   bundleSize
+#   bundleId
+#   recordId
+#   serialNumber
+# (not included) validSignature
+##################################
+
+# Value constraints
+LATITUDE_UPPER_LIMIT = Decimal(90.0)
+LATITUDE_LOWER_LIMIT = Decimal(-90.0)
+LONGITUDE_UPPER_LIMIT = Decimal(180.0)
+LONGITUDE_LOWER_LIMIT = Decimal(-180.0)
+ELEVATION_UPPER_LIMIT = Decimal(6143.9)
+ELEVATION_LOWER_LIMIT = Decimal(-409.6)
+SPEED_UPPER_LIMIT = Decimal(163.82)
+SPEED_LOWER_LIMIT = Decimal(0)
+HEADING_UPPER_LIMIT = Decimal(359.9875)
+HEADING_LOWER_LIMIT = Decimal(0)
+BSM_SOURCE_LIST = ['RV', 'EV', 'unknown']
+# BSM_SOURCE_LIST = ['bob','alice']
+RX_SOURCE_LIST = ['RSU', 'SAT', 'RV', 'SNMP', 'NA', 'unknown']
+RECORD_GENERATED_BY_LIST = ['TMC', 'OBU', 'RSU', 'TMC_VIA_SAT', 'TMC_VIA_SNMP']
+RECORD_TYPE_LIST = ['bsmLogDuringEvent', 'rxMsg', 'dnMsg', 'bsmTx', 'driverAlert', 'unsupported']
+SANITIZED_LIST = [True, False]
+SCHEMA_VERSION = 6
+SECURITY_RESULT_CODE_LIST = ['success', 'unknown', 'inconsistentInputParameters', 'spduParsingInvalidInput', 'spduParsingUnsupportedCriticalInformationField', 'spduParsingCertificateNotFound', 'spduParsingGenerationTimeNotAvailable', 'spduParsingGenerationLocationNotAvailable', 'spduCertificateChainNotEnoughInformationToConstructChain', 'spduCertificateChainChainEndedAtUntrustedRoot', 'spduCertificateChainChainWasTooLongForImplementation', 'spduCertificateChainCertificateRevoked', 'spduCertificateChainOverdueCRL', 'spduCertificateChainInconsistentExpiryTimes', 'spduCertificateChainInconsistentStartTimes', 'spduCertificateChainInconsistentChainPermissions', 'spduCryptoVerificationFailure', 'spduConsistencyFutureCertificateAtGenerationTime', 'spduConsistencyExpiredCertificateAtGenerationTime', 'spduConsistencyExpiryDateTooEarly', 'spduConsistencyExpiryDateTooLate', 'spduConsistencyGenerationLocationOutsideValidityRegion', 'spduConsistencyNoGenerationLocation', 'spduConsistencyUnauthorizedPSID', 'spduInternalConsistencyExpiryTimeBeforeGenerationTime', 'spduInternalConsistencyextDataHashDoesntMatch', 'spduInternalConsistencynoExtDataHashProvided', 'spduInternalConsistencynoExtDataHashPresent', 'spduLocalConsistencyPSIDsDontMatch', 'spduLocalConsistencyChainWasTooLongForSDEE', 'spduRelevanceGenerationTimeTooFarInPast', 'spduRelevanceGenerationTimeTooFarInFuture', 'spduRelevanceExpiryTimeInPast', 'spduRelevanceGenerationLocationTooDistant', 'spduRelevanceReplayedSpdu', 'spduCertificateExpired']
+BUNDLE_SIZE_UPPER_LIMIT = 2147483648
+BUNDLE_SIZE_LOWER_LIMIT = 1
+BUNDLE_ID_UPPER_LIMIT = 9223372036854775807
+BUNDLE_ID_LOWER_LIMIT = 0
+RECORD_ID_UPPER_LIMIT = 2147483647
+RECORD_ID_LOWER_LIMIT = 0
+SERIAL_NUMBER_UPPER_LIMIT = 9223372036854775807
+SERIAL_NUMBER_LOWER_LIMIT = 0
+
+# validates all fields of the metadata
+def validate_metadata(metadata, errored):
+    validate_field(errored, 'payloadType', metadata.get('payloadType'))
```

Comment:
All these calls to `validate_medatdata` should ideally be made in a loop. The list that is looped through should come from the config file or from the combination of config, input file and expected output file.